src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/talismans/MagicianTalisman.java

```diff
@@ -40,12 +41,17 @@ public MagicianTalisman(SlimefunItemStack item, ItemStack[] recipe) {
         addItemSetting(allowEnchantmentBooks);
 
         for (Enchantment enchantment : Enchantment.values()) {
-            try {
-                for (int i = 1; i <= enchantment.getMaxLevel(); i++) {
-                    enchantments.add(new TalismanEnchantment(enchantment, i));
+            // Stop custom enchant plugins appearing
+            // We also want to stop some stupid plugins which register as Minecraft enchants
+            // but also go above the max enchant level.
+            if (enchantment.getKey().getNamespace().equals(NamespacedKey.MINECRAFT) && enchantment.getMaxLevel() < Short.MAX_VALUE) {
```

Comment:
I think you mixed this one up.
This will exclude any enchantment registering under their own namespace though.
So only `minecraft:**` enchantments are supported now, which doesn't fix the issue of faulty plugins registering theirs under the wrong namespace.

Reply:
It's probably best to not support custom enchants either way honestly. They can be buggy but yeah, the second check fixes if they register under MC and messed up max level.

I can remove the first check and allow custom enchants I guess but it can still have issues.
For example, none of these custom enchants will actually show up on books or items. So, it can look like a book has no enchant when there's actually a custom one there.

I think for the sake of simplicity it's best to just make it only vanilla.

Reply:
Or perhaps a config option for it.
And then have it disabled by default and if someone asks about their favourite enchantment plugin, we can tell them to enable this setting but warn of the consequences.

Reply:
Yeah that can work. 

I hate custom enchant plugins though. Also hate Bukkit for having getMaxLevel as an int rather than short.

Reply:
same and same.
But then again... very typical thing for Bukkit :LUL: